odoo/models.py

```diff
@@ -2608,12 +2607,11 @@ def qualify(field):
         # determine the actual query to execute
         from_clause, where_clause, params = query.get_sql()
         query_str = """ SELECT %(qual_names)s FROM %(from_clause)s
-                        WHERE %(where_clause)s %(order_str)s
```

Comment:
> is the field ordering already determined when the metaclass gets them

That one, it's collected beforehand and the metaclass only gets a (regular, unordered) mapping. [Python 3 adds a `__prepare__` hook which can return a custom mapping](https://www.python.org/dev/peps/pep-3115/#invoking-the-metaclass) but that's not available in P2.

The option in P2 would be to do something similar to what Django does with [a counter on the fields](https://github.com/django/django/blob/stable/1.11.x/django/forms/fields.py#L116) and [sorting based on that counter](https://github.com/django/django/blob/stable/1.11.x/django/forms/forms.py#L40) when creating the class.